services/venice-controller/src/main/java/com/linkedin/venice/controller/init/SystemStoreInitializationHelper.java

```diff
@@ -161,6 +162,11 @@ public static void setupSystemStore(
     }
 
     if (updateStoreQueryParams != null && updateStoreCheckSupplier.apply(store)) {
+      if (store.getPartitionCount() == 0 && !updateStoreQueryParams.getPartitionCount().isPresent()) {
+        updateStoreQueryParams
+            .setPartitionCount(multiClusterConfigs.getControllerConfig(clusterName).getMinNumberOfPartitions());
+      }
+
```

Comment:
What if updateStoreQueryParams exists and the partition count is set to invalid count? e.g., <= minNumberOfPartitions? Is that possible?

Reply:
This will get validated in `UpdateStoreUtils#validateStorePartitionCountUpdate`